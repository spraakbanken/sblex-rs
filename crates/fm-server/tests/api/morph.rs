use reqwest::StatusCode;

use crate::conftest::TestApp;

#[tokio::test]
async fn can_call_morph_w_cont() -> eyre::Result<()> {
    // Arrange
    let app = TestApp::new().await?;
    let client = reqwest::Client::new();

    // Act
    let response = client.get(app.url("/morph-w-cont/dväljes")?).send().await?;

    // Assert
    let status_code = response.status();

    assert_eq!(status_code, StatusCode::OK);
    let data: serde_json::Value = response.json().await?;
    insta::assert_json_snapshot!(data);

    Ok(())
}

#[tokio::test]
async fn can_call_morph() -> eyre::Result<()> {
    // Arrange
    let app = TestApp::new().await?;
    let client = reqwest::Client::new();

    // Act
    let response = client.get(app.url("/morph/dväljes")?).send().await?;

    // Assert
    let status_code = response.status();

    assert_eq!(status_code, StatusCode::OK);
    let data: serde_json::Value = response.json().await?;
    insta::assert_json_snapshot!(data);

    Ok(())
}

#[tokio::test]
async fn non_existent() -> eyre::Result<()> {
    // Arrange
    let app = TestApp::new().await?;
    let client = reqwest::Client::new();

    // Act
    let response = client.get(app.url("/morph/löparsko")?).send().await?;

    // Assert
    let status_code = response.status();

    assert_eq!(status_code, StatusCode::NOT_FOUND);

    // let data: serde_json::Value = response.json().await?;
    // insta::assert_json_snapshot!(data);

    Ok(())
}
