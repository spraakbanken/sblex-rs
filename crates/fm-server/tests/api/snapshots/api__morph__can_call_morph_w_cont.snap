---
source: crates/fm-server/tests/api/morph.rs
expression: data
---
{
  "a": [
    {
      "gf": "dväljas",
      "id": "dväljas..vb.1",
      "is": [],
      "msd": "pres ind s-form",
      "p": "vb_vs_dväljas",
      "pos": "vb"
    },
    {
      "gf": "dväljas",
      "id": "dväljas..vb.1",
      "is": [],
      "msd": "imper",
      "p": "vb_vs_dväljas",
      "pos": "vb"
    }
  ],
  "c": ""
}
